azurelinuxagent/common/protocol/restapi.py

```diff
@@ -68,9 +68,9 @@ def __init__(self):
         self.certificates = DataContractList(Cert)
 
 
-class VMAgentManifest(object):
-    def __init__(self, family, version=None):
-        self.family = family
+class VMAgentFamily(object):
```

Comment:
I renamed VMAgentManifest to VMAgentFamily because this class does not represent a manifest, but rather the GAFamilies element in ExtensionConfigs (which includes the name of the family (Prod vs Test), the required agent version, and the URIs of the actual manifests). This naming made the code extremely confusing, especially in code that deals with VMAgentManifest and actual agent manifest. You will see those renames throughout this PR.